### DISCUSSION TITLE
How to compute SHA1 hash with openpgp.js v5?

### DISCUSSION THREAD

**Discussion by IvanPizhenko** [openpgpjs/openpgpjs#1515] [Q&A]
Previously, in v4, openpgp.js was exporting some hashing functions and following code was possible:
```typescript
await opgp.crypto.hash.digest(opgp.enums.hash.sha1, Buf.fromUtfStr(user.toLowerCase()))
```
Now function `digest()` is not exported. What is recommended way to compute hash now?

**Comment by twiss** [comment]
Hey :wave: The recommended way is to use the native crypto API of the platform you're using, i.e. https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest in the browser or https://nodejs.org/api/crypto.html#class-hash on Node.js.
